tests/api_connexion/endpoints/test_extra_link_endpoint.py

```diff
@@ -15,24 +15,196 @@
+        )
+
+        self.assertEqual(200, response.status_code, response.data)
+        self.assertEqual(
+            {
+                "BigQuery Console": "https://console.cloud.google.com/bigquery?j=TEST_JOB_ID",
+            },
+            response.json,
+        )
+
+    @mock_plugin_manager(plugins=[])
+    def test_should_response_200_missing_xcom(self):
+        response = self.client.get(
+            "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_SINGLE_QUERY/links"
+        )
+
+        self.assertEqual(200, response.status_code, response.data)
+        self.assertEqual(
+            {
+                "BigQuery Console": None,
+            },
+            response.json,
+        )
+
+    @mock_plugin_manager(plugins=[])
+    def test_should_response_200_multiple_links(self):
+        XCom.set(
+            key="job_id",
+            value=["TEST_JOB_ID_1", "TEST_JOB_ID_2"],
+            execution_date=self.now,
+            task_id="TEST_MULTIPLE_QUERY",
+            dag_id=self.dag.dag_id,
+        )
+        response = self.client.get(
+            "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_MULTIPLE_QUERY/links"
+        )
+
+        self.assertEqual(200, response.status_code, response.data)
+        self.assertEqual(
+            {
+                "BigQuery Console #1": "https://console.cloud.google.com/bigquery?j=TEST_JOB_ID_1",
+                "BigQuery Console #2": "https://console.cloud.google.com/bigquery?j=TEST_JOB_ID_2",
+            },
+            response.json,
+        )
+
+    @mock_plugin_manager(plugins=[])
+    def test_should_response_200_multiple_links_missing_xcom(self):
         response = self.client.get(
-            "/dags/TEST_DG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_TASK_ID/links"
+            "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_MULTIPLE_QUERY/links"
         )
-        assert response.status_code == 200
+
+        self.assertEqual(200, response.status_code, response.data)
+        self.assertEqual(
+            {
+                "BigQuery Console #1": None,
+                "BigQuery Console #2": None,
+            },
+            response.json,
+        )
+
+    def test_should_response_200_support_plugins(self):
+        class GoogleLink(BaseOperatorLink):
+            name = "Google"
+
+            def get_link(self, operator, dttm):
+                return "https://www.google.com"
+
+        class S3LogLink(BaseOperatorLink):
+            name = 'S3'
+            operators = [BigQueryExecuteQueryOperator]
+
+            def get_link(self, operator, dttm):
+                return 'https://s3.amazonaws.com/airflow-logs/{dag_id}/{task_id}/{execution_date}'.format(
+                    dag_id=operator.dag_id,
+                    task_id=operator.task_id,
+                    execution_date=quote(dttm.isoformat()),
```

Comment:
```suggestion
                    execution_date=quote_plus(dttm.isoformat()),
```
I am thinking that quote_plus will be better here?

Reply:
```2020-01-01T00:00:00+00:00```
It doesn't make a difference here because we don't have spaces.